### DISCUSSION TITLE
ViewNode using after 2.1.0

### DISCUSSION THREAD

**Discussion by rawello** [sceneview/sceneview#471] [Q&A]
Hello everyone, i cant understand how create ViewNode after that update. My app wroted at sceneform, after i used 0.7.0 sceneview and 1.2.0 but in that update i broke my brain. Sorry for dunkey question but i cant create ViewNode as object(i use fragments and ARSceneView).

**Comment by rawello** [comment]
update my question: i creating ViewNode, but getting black View(not rendered) https://drive.google.com/file/d/1Dd8VmHjhMH-2BypYYFED9OGxy9_B_Qha/view?usp=drivesdk. Trying using sample from https://github.com/SceneView/sceneview-android/pull/461 but same result

**Comment by ThomasGorisse** [comment]
ViewNode is available and improved in v4.0.0-rc.1:

```kotlin
SceneView {
    ViewNode(
        viewFactory = { context ->
            // Any Android View
            TextView(context).apply { text = "Hello 3D!" }
        },
        position = Position(0f, 1f, 0f)
    )
}
```

A recent fix also resolved a lifecycle issue where ViewNode could show a black rectangle after activity resume. Make sure to use the latest v4.0.0-rc.1.
